### ISSUE TITLE
Question "Does the stop have a curb on the side of the road?" is not-answerable when there is no road.

### ISSUE THREAD

**Issue by dreirund** [OPENER-next/OpenStop#278]
### Problem

There is the question "Does the stop have a curb on the side of the road?". But I encounter tram stops where there is no road; the tram runs on it's own bed.

The question assumes a road but this assumption is not always true.

I do not know how to answer ("yes" because the platform is higher than the rail bed? "No" because there is no road?).

This question should either be more clear to include cases where there is no road, or add a third option "there is no road".

### Which app version do you use?

0.8.1+20

### What operating system do you use?

Android

### What operating system *version* do you use?

11

**Comment by wielandb** [comment]
This needs some more discussion about what we are actually asking about.

My first impulse was to just omit the word "road", because no matter what, there is a measureable height difference between the stop and whatever it is the stop is next to. On the other hand, we ask this so we know what the situation is like for busses and wheelchair users at this stop. At a tram stop, we do not really need information about the kerb height, because if the entering of a tram is level and possible with a wheelchair depends more on the tram itself than on the kerb... 

Idea for solution: Do not ask the kerb height question if `tram=` is the only vehicle stopping at a stop.

**Comment by dreirund** [comment]
> because if the entering of a tram is level and possible with a wheelchair depends more on the tram itself than on the kerb...

Which is also the case for quite a couple of bus stops, see e.g. "[Dresdner Combiboard](https://www.dvb.de/de-de/die-dvb/technik/gleissysteme/combibord)".

> Idea for solution: Do not ask the kerb height question if `tram=` is the only vehicle stopping at a stop.

I don't see that the app has a solution to this already, I am still presented with the question where I have no idea how to answer this.

**Comment by Robbendebiene** [comment]
This issue should largely be mitigated. It would be helpful if you could provide a link to an OSM element where this question is asked.

I haven't looked into it again but if I remember correctly the current logic is:
- Ask the question at `railway=platform` with `bus=yes` (assuming it is a tram and bus stop)
- Ask the question at `highway=platform` assuming it is at least bus stop (and perhaps a tram stop as well if `tram=yes`)

The latter condition is probably the one usually causing issues, but then this is a mapping issue, at least as far as I know a tram only platform should be mapped with `railway=platform`

**Comment by dreirund** [comment]
> *This issue should largely be mitigated. It would be helpful if you could provide a link to an OSM element where this question is asked.*

For example, [&nearr;&hairsp;here](https://www.openstreetmap.org/way/1085463403).

> *I haven't looked into it again but if I remember correctly the current logic is:*
> 
>     * *Ask the question at `railway=platform` with `bus=yes` (assuming it is a tram and bus stop)*
>     * *Ask the question at `highway=platform` assuming it is at least bus stop (and perhaps a tram stop as well if `tram=yes`)*
> 
> *The latter condition is probably the one usually causing issues, but then this is a mapping issue, at least as far as I know a tram only platform should be mapped with `railway=platform`*

Yes, it is the latter one: `highway=platform`. A tram-only platform not at a road, but also part of useful foot walking when getting from A to B. (And the [dedicated footway along it](https://www.openstreetmap.org/way/1085907617) was only added later after recognising that some routing engine(s) did not route along the platform.)

OK, I will read up the tags and see if this should be changed; feel free to also leave comments about this here or as a comment to the [commit introducing `highway=platform`](https://www.openstreetmap.org/way/1085463403/history/3).

Regards!

**Comment by Robbendebiene** [comment]
Thanks for digging into this. I admit that it is not 100% clear




> The usage of other tags is similar to railways:
>  [...]
    platforms: [public_transport](https://wiki.openstreetmap.org/wiki/Key:public_transport)=[platform](https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dplatform) + [railway](https://wiki.openstreetmap.org/wiki/Key:railway)=[platform](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dplatform).
>  [...]
> If the stop is a dual bus + tram stop, a single [public_transport](https://wiki.openstreetmap.org/wiki/Key:public_transport)=[platform](https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dplatform) object can be shared by both transport modes. In that case if a physical platform exists, the [railway](https://wiki.openstreetmap.org/wiki/Key:railway)=[platform](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dplatform) tag can be added to the [highway](https://wiki.openstreetmap.org/wiki/Key:highway)=[platform](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dplatform) way instead.

**Comment by dreirund** [comment]
> *This kinda states it can be used for trams. On the other hand it states "for platforms (raised structures) along the side of a road" which at least in our "kerb question" case would be okay.*
> 
> *If we come to the conclusion that it is indeed correct to map tram platforms **not** nearby a road with `highway=platform`*

I came to the conclusion that `highway=platform` is only to where it really is along the road.

I also drew from another already mapped example in the same town, additionally to the mentioned OSM wiki sites.

So I close this issue now as invalid, because wrong tagging did lead to it.

However, it can be helpful that, when the user of OpenStop finds something wrongly tagged, to leave a note or to un-tag the wrong tag (in special cases; as StreetComplete allows for changing surface types for example, and maybe also "track" and "path"? I don't know).

**Comment by dreirund** [comment]
Two more questions:

1. Is this really an accessibility question about how to board the bus? Since there are many bus stops with raised platform around which are built _specifically_ such that the bus can stop directly at the platform so that a wheelchair can roll in easily, without a step and a very small gap. But yes, the platform is raised from the road level to achieve this.  
   So, if this is an accessibility question, it is confusing. Since I would answer a height of e.g. 20cm (have not measured yet), but exactly that makes it wheelchair accessivle to board the bus.
2. There are [platforms where at one side there are only tram tracks and no road, on the other side a road and a bus stop](https://www.openstreetmap.org/way/217916785). In that case, answer the question for the road facing side of the platform and ignore the tram facing side?

Regards!

**Comment by Robbendebiene** [comment]
> Is this really an accessibility question about how to board the bus? Since there are many bus stops with raised platform around which are built specifically such that the bus can stop directly at the platform so that a wheelchair can roll in easily, without a step and a very small gap. But yes, the platform is raised from the road level to achieve this.
So, if this is an accessibility question, it is confusing. Since I would answer a height of e.g. 20cm (have not measured yet), but exactly that makes it wheelchair accessivle to board the bus.

This question just resembles a different level of detail for the kerb/platform height question. If you know there is a no kerb and the bus stops at the platform you know for sure that you cannot enter the bus step-less. If the entry point into a tram is on the road and the platform does not have a lowered kerb you also know that there will be a problem. So you can derive some useful information from it.
Of course we would prefer the actual height (there is a question for that, but measuring is hard and in this case also dangerous wherefore it is part of the advanced question catalogue).

> There are [platforms where at one side there are only tram tracks and no road, on the other side a road and a bus stop](https://www.openstreetmap.org/way/217916785). In that case, answer the question for the road facing side of the platform and ignore the tram facing side?

Some time ago we discussed, whether to ask the kerb question on a platform mapped as an area or not. Following the interpretation from the [wiki](https://wiki.openstreetmap.org/wiki/Key:kerb) we decided that there is no meaningful difference to asking it for a platform mapped as a way or node. The [wiki](https://wiki.openstreetmap.org/wiki/Key:kerb) states (unfortunately only in the example section):

> It is assumed the kerb tag refers to the edge between the road and the platform.

So the answer to your question would be yes 🙂

**Comment by dreirund** [comment]
> > Is this really an accessibility question about how to board the bus? Since there are many bus stops with raised platform around which are built specifically such that the bus can stop directly at the platform so that a wheelchair can roll in easily, without a step and a very small gap. But yes, the platform is raised from the road level to achieve this.
> > So, if this is an accessibility question, it is confusing. Since I would answer a height of e.g. 20cm (have not measured yet), but exactly that makes it wheelchair accessivle to board the bus.
> 
> This question just resembles a different level of detail for the kerb/platform height question.

Yes, I figured out that _after_ answering the kerb height question there was a question if the kerb is special to aid busses approaching the platform. Have I played through it earlier it would have came clear to me by myself, because the second question then makes it clear for me that _this_ actually is the accessibility related one.

Regards!
